src/bitcoin-mine.cpp

```diff
@@ -0,0 +1,116 @@
+    init::AddLoggingArgs(args);
+}
+
+MAIN_FUNCTION
+{
+    ArgsManager& args = gArgs;
+    AddArgs(args);
+    std::string error_message;
+    if (!args.ParseParameters(argc, argv, error_message)) {
+        tfm::format(std::cerr, "Error parsing command line arguments: %s\n", error_message);
+        return EXIT_FAILURE;
+    }
+    if (!args.ReadConfigFiles(error_message, true)) {
+        tfm::format(std::cerr, "Error reading config files: %s\n", error_message);
+        return EXIT_FAILURE;
+    }
+    if (HelpRequested(args) || args.IsArgSet("-version")) {
+        std::string output{strprintf("%s bitcoin-mine version", PACKAGE_NAME) + " " + FormatFullVersion() + "\n"};
+        if (args.IsArgSet("-version")) {
+            output += FormatParagraph(LicenseInfo());
+        } else {
+            output += HELP_USAGE;
+            output += args.GetHelpMessage();
+            output += HELP_EXAMPLES;
+        }
+        tfm::format(std::cout, "%s", output);
+        return EXIT_SUCCESS;
+    }
+    if (!CheckDataDirOption(args)) {
+        tfm::format(std::cerr, "Error: Specified data directory \"%s\" does not exist.\n", args.GetArg("-datadir", ""));
+        return EXIT_FAILURE;
+    }
+    SelectParams(args.GetChainType());
+
+    // Set logging options but override -printtoconsole default to depend on -debug rather than -daemon
+    init::SetLoggingOptions(args);
+    LogInstance().m_print_to_console = args.GetBoolArg("-printtoconsole", args.GetBoolArg("-debug", false));
+    if (!init::StartLogging(args)) {
+        tfm::format(std::cerr, "Error: StartLogging failed\n");
+        return EXIT_FAILURE;
+    }
+
+    // Connect to existing bitcoin-node process or spawn new one.
+    std::unique_ptr<interfaces::Init> mine_init{interfaces::MakeMineInit(argc, argv)};
+    assert(mine_init);
+    std::unique_ptr<interfaces::Init> node_init;
+    try {
+        std::string address{args.GetArg("-ipcconnect", "unix")};
+        node_init = mine_init->ipc()->connectAddress(address);
+    } catch (const std::exception& exception) {
+        tfm::format(std::cerr, "Error: %s\n", exception.what());
+        tfm::format(std::cerr, "Probably bitcoin-node is not running or not listening on a unix socket. Can be started with:\n\n");
+        tfm::format(std::cerr, "    bitcoin-node -chain=%s -ipcbind=unix\n", args.GetChainTypeString());
+        return EXIT_FAILURE;
+    }
+    assert(node_init);
+    tfm::format(std::cout, "Connected to bitcoin-node\n");
+    std::unique_ptr<interfaces::Mining> mining{node_init->makeMining()};
+    assert(mining);
+
+    auto tip{mining->getTip()};
+    if (tip) {
+        tfm::format(std::cout, "Tip hash is %s.\n", tip->hash.ToString());
+    } else {
+        tfm::format(std::cout, "Tip hash is null.\n");
```

Comment:
I can't reach this condition. Even when I spin up a mainnet node and immediately start `bitcoin-mine` it just seems to wait a bit and then return a hash.

Maybe we should simplify the `getTip()` interface to not return an optional. Though I'm not sure if it's completely impossible to hit this.